packages/google_maps_flutter/lib/src/controller.dart

```diff
@@ -198,10 +198,28 @@ class GoogleMapController extends ChangeNotifier {
   Future<void> removeMarker(Marker marker) async {
     assert(marker != null);
     assert(_markers[marker._id] == marker);
-    await _channel.invokeMethod('marker#remove', <String, dynamic>{
-      'marker': marker._id,
+    await _removeMarker(marker._id);
+    notifyListeners();
+  }
+
+  /// Removes all [markers] from the map.
+  ///
+  /// Change listeners are notified once all markers have been removed on the
+  /// platform side.
+  ///
+  /// The returned [Future] completes once listeners have been notified.
+  Future<void> clearMarkers() async {
+    assert(_markers != null);
+    _markers.forEach((String id, Marker marker) async {
```

Comment:
I think that nothing awaits the closure passed to forEach, so the cleanMarkers the notifyListeners will happen immediately and the clearMarkers method will complete immediately...